csharp-intermediate-topics/WhenToUseReaderWriterLockSlimOverLockInCSharp/WhenToUseReaderWriterLockSlimOverLockInCSharp/Program.cs

```diff
@@ -0,0 +1,22 @@
+﻿using WhenToUseReaderWriterLockSlimOverLockInCSharp;
+
+var config = new ThreadExecutionConfiguration
+{
+    ReaderThreadsCount = 5,
+    WriterThreadsCount = 1,
+    ReaderExecutionDelay = 100,
+    WriterExecutionDelay = 100,
+    ReaderExecutionsCount = 100000,
+    WriterExecutionsCount = 100000,
+};
+
+var lockReadWrite = new LockReadWrite();
+Console.WriteLine($"LockReadWrite execution time: {lockReadWrite.Execute(config)} milliseconds.");
+
+var readerWriterLockReadWrite = new ReaderWriterLockReadWrite();
+Console.WriteLine($"ReaderWriterLock execution time: {readerWriterLockReadWrite.Execute(config)} milliseconds.");
+
+var readerWriterLockSlimReadWrite = new ReaderWriterLockSlimReadWrite();
+Console.WriteLine($"ReaderWriterLockSlim execution time: {readerWriterLockSlimReadWrite.Execute(config)} milliseconds.");
+
+Console.ReadKey();
```

Comment:
This is not needed. If we are running in an IDE, this adds the "press a key" for us. If running on the command line, no need to pause as the text will be there for us.